marshal/tests/serialization/set.go

```diff
@@ -0,0 +1,104 @@
+package serialization
+
+import (
+	"bytes"
+	"fmt"
+	"runtime/debug"
+	"testing"
+
+	"github.com/gocql/gocql/marshal/tests/funcs"
+	"github.com/gocql/gocql/marshal/tests/utils"
+)
+
+type Sets []*Set
+
+// Set is a tool for generating test cases of marshal and unmarshall funcs.
+// For cases when the function should no error,
+// marshaled data from Set.Values should be equal with Set.Data,
+// unmarshalled value from Set.Data should be equal with Set.Values.
+type Set struct {
+	Data   []byte
+	Values []interface{}
+
+	IssueMarshal   string
+	IssueUnmarshal string
+}
+
+func (s Set) Run(name string, t *testing.T, marshal func(interface{}) ([]byte, error), unmarshal func([]byte, interface{}) error) {
+	if name == "" {
+		name = utils.ValueToName(s.Values[0])
+	}
+
+	t.Run(name, func(t *testing.T) {
+		for i := range s.Values {
+			val := s.Values[i]
+
+			t.Run(fmt.Sprintf("%T", val), func(t *testing.T) {
```

Comment:
```suggestion
			t.Run(utils.ValueToName(modVal), func(t *testing.T) {
```

Where `utils.ValueToName` is:
```
func ValueToName(val interface{}) string {
	return fmt.Sprintf("%T(\"%v\")", val, unwrapPointer(val))
}

func unwrapPointer(val interface{}) interface{} {
	rv := reflect.ValueOf(val)
	for rv.Kind() == reflect.Ptr {
		if rv.IsNil() {
			return "nil"
		}
		rv = rv.Elem()
	}
	return rv.Interface()
}
```

Reply:
This is a complication that is fraught with problems with the transformation of different types into a string, which have already been explained. 
In addition, it does not take into account, that in each `Set` only types are dynamic, everything else is static, even the values of these types.
The including something static for the whole Set into the name of t.Run() for each value - making a testing report more unreadable.

Reply:
My bad, I did not provide whole thing, my proposal to do something like this, there is no static thing in the test name:
```
	modsWithRaw := mod.Mods{mod.Raw}
	modsWithRaw = append(modsWithRaw, mods...)
	for _, m := range modsWithRaw {
		for _, modVal := range m.Apply(s.Values) {
			t.Run(utils.ValueToName(modVal), func(t *testing.T) {
```